nix-darwin/default.nix

```
@@ -55,6 +58,24 @@ in
         '';
       };
 
+      extraSpecialArgs = mkOption {
+        type = types.attrs;
+        default = { };
+        example = literalExample "{ modulesPath = ../modules; }";
+        description = ''
+          Extra <literal>specialArgs</literal> passed to Home Manager.
+        '';
+      };
+
+      extraModules = mkOption {
+        type = with types; listOf (oneOf [ attrs (functionTo attrs) path ]);
+        default = [ ];
+        example = literalExample "[ { home.packages = [ nixpkgs-fmt ]; } ]";
+        description = ''
+          Extra modules added to all users.
+        '';
+      };
```

```suggestion
      extraDefaultModules = mkOption {
        type = with types; listOf (oneOf [ attrs (functionTo attrs) path ]);
        default = [ ];
        example = literalExample "[ { home.packages = [ nixpkgs-fmt ]; } ]";
        description = ''
          Extra modules added to all users.
        '';
      };
```

What do you think of this name?

Same for the one used in `nixos/default.nix`.

---

I don't mind it, but its a little confusing to me. Default usually means this option is the default *to* something. But these are the only modules set and all of them get added. I'm not entirely against it, but it feels a bit weird. 

I do like the name the other PR used, sharedModules.

---

Let's go for `sharedModules` then

---

sounds good, changed it